pkg/server/internal_server.go

```diff
@@ -0,0 +1,31 @@
+package server
+
+import (
+	"flag"
+	"time"
+
+	serverww "github.com/weaveworks/common/server"
+)
+
+// Config extends weaveworks server config
+type Config struct {
+	serverww.Config `yaml:",inline"`
```

Comment:
That also includes a lot of struct fields which are unused, I would prefer to shrink those down a bit more.

I don't know if it even needs to be supporting TLS, but I guess the list of the registered flags would be a good start.

Reply:
In the scope of exposing only the ready handler it makes sense to shrink this more. However, as you mention above it makes sense to expose the metrics from this handler too and thus I would like to keep the inline struct as such for start but limit the inputs via the shrinked registered flags. As per exposing metrics allowing TLS from start is imho a good thing to do. For example on OpenShift we allow only scraping over HTTPS.